bokehjs/src/lib/models/glyphs/image.ts

```diff
@@ -25,9 +25,9 @@ export class ImageView extends ImageBaseView {
     }
   }
 
-  protected _flat_img_to_buf8(img: Arrayable<number>): Uint8ClampedArray {
+  protected _flat_img_to_buf8(img: Arrayable<number>, length_divisor: number): Uint8ClampedArray {
```

Comment:
This `length_divisor` isn't ideal, but is the easiest way to add the new functionality here. Really it is the length of the third dimension of the supplied array, but as we are dropping the dimensionality of arrays before they get here the information is not available so has to be passed in from higher up the call stack.

Reply:
We need to start using ndarrays in such APIs, but that's something for another time, as the existing ndarrays in bokehjs aren't that useful right now.